gateway/uplink.go

```diff
@@ -24,13 +25,19 @@ const (
 	Confirmed   uplinkType = "confirmed"
 )
 
+var errInvalidLength = errors.New("unexpected payload length")
+
 // Structure of phyPayload:
 // | MHDR | DEV ADDR|  FCTL |   FCnt  | FPort   |  FOpts     |  FRM Payload | MIC |
 // | 1 B  |   4 B    | 1 B   |  2 B   |   1 B   | variable    |  variable   | 4B  |
 // Returns the node name, readings and error.
 func (g *gateway) parseDataUplink(ctx context.Context, phyPayload []byte, packetTime time.Time, snr float64, sf int) (
 	string, map[string]interface{}, error,
 ) {
+	if len(phyPayload) < 13 {
+		return "", map[string]interface{}{}, errInvalidLength
+	}
+
```

Comment:
Please `erros.Join()` the two returned errors in this file with a descriptive error that tells us in what part of the stack we're seeing the invalid payloads, for better insights when they do happen.